glyphs-reader/src/font.rs

```diff
@@ -1502,17 +1502,54 @@ fn default_master_idx(raw_font: &RawFont) -> usize {
                 .enumerate()
                 .find(|(_, master)| master.id == origin)
                 .map(|(idx, _)| idx)
-                .unwrap_or(0)
         })
-        // TODO: implement searching for "a base style shared between all masters" as glyphsLib does
-        // Still nothing? - just look for one called Regular
-        .or_else(|| {
-            raw_font
-                .font_master
-                .iter()
-                .position(|m| matches!(m.name.as_deref(), Some("Regular")))
+    {
+        return master_idx;
+    }
+    // Look for a master whose style is shared by all masters, defaulting to 'Regular'
```

Comment:
As below, this is factually correct but doesn't really help understand the rationale.

My kneejerk reaction is the code feels more complicated than the underlying operation ... let me think about that a bit. Probably more helpful if it comes with an actual suggestion :)

Reply:
https://github.com/googlefonts/fontc/pull/1001 attempts to:

1. Implement in terms of picking the best master
1. Expand comments based on parts I found hard to follow
1. Operate on `Vec<&str>` rather than forming String